viewregistry-android/src/main/java/com/squareup/viewregistry/NoEffect.kt

```diff
@@ -0,0 +1,67 @@
+package com.squareup.viewregistry
+
+import android.view.View
+import android.view.ViewGroup
+import com.squareup.viewregistry.BackStackScreen.Key
+import com.squareup.viewregistry.ViewStateStack.Direction
+import com.squareup.viewregistry.ViewStateStack.UpdateTools
+import io.reactivex.Observable
+import kotlin.reflect.KClass
+
+/**
+ * Empties the container and adds the view to it with no animation.
+ * This is the default effect returned by [ViewRegistry.effects]
+ * if no other is found.
+ */
+class NoEffect(
+  private val from: Key<*>?,
+  private val to: Key<*>?,
+  private val direction: Direction?
+) : BackStackEffect {
+  constructor(
+    from: KClass<*>? = null,
+    to: KClass<*>? = null,
+    direction: Direction? = null
+  ) : this(from?.let { Key(it) }, to?.let { Key(it) }, direction)
+
+  override fun matches(
+    from: Key<*>,
+    to: Key<*>,
+    direction: Direction
+  ): Boolean {
+    return (this.from == null || this.from == from) &&
+        (this.to == null || this.to == to) &&
+        (this.direction == null || this.direction == direction)
+  }
+
+  override fun execute(
+    from: View,
+    to: BackStackScreen<*>,
+    screens: Observable<out BackStackScreen<*>>,
+    viewRegistry: ViewRegistry,
+    container: ViewGroup,
+    tools: UpdateTools
+  ) {
+    tools.saveOldView(from)
```

Comment:
It's because `TransitionManager` wants to create the view for us, and some effects might want to use `TransitionManager`. Therefore all effects have to be responsible for creating the views, and therefore all effect have to be responsible for apply the save and restore functions.

Reply:
Well, I guess that only explains the `restore()` call. Trying to remember why they both wound up in here, thinking…

Reply:
You're right, this is silly. I'll follow up.